ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/sqlstd/SQLAuthorizationUtils.java

```diff
@@ -271,7 +271,16 @@ private static boolean isOwner(IMetaStoreClient metastoreClient, String userName
         thriftTableObj = metastoreClient.getTable(hivePrivObject.getDbname(),
             hivePrivObject.getObjectName());
       } catch (Exception e) {
-        throwGetObjErr(e, hivePrivObject);
+        boolean isTableExists = true;
+        try {
+          if(!metastoreClient.tableExists(hivePrivObject.getDbname(), hivePrivObject.getObjectName())) {
```

Comment:
it would be better to place this check at the call site-s to this method - so that the explanation will not be needed at all

Reply:
I think this method should throw an exception if the table doesn't exists - you can't own non-exiting items; that's misleading...

* this is a private method - so it could only be called from this class
* the `metastoreClient` is passed as an argument to this method - so I don't think that should be a problem...